---
title: "Pv_Hep_scRNAseq_additional_files_4"
author: "Anthony Ruberto"
date: "`r Sys.Date()`"
output: html_document
---

Here we will perform the differential gene expression analysis used in the
manuscript for the parasite side.


```{r libraries}
library(Seurat)
```

Let's load the Seurat objects.
```{r load Seurat objects}
# Object used for Schizont versus Hypnozoite comparison
readRDS(Hep59.1.2.seu, "outputs/Hep59.1.2.seu.rds")
# Object used for Hypnozoite only comparison
readRDS(Hypnozoites, "outputs/Hypnozoites.rds")
```

DE testing.
```{r DE testing schizonts verus hypnozoites}
DefaultAssay(Hep59.1.2.seu) <- "RNA"
Hep59.1.2.seu<-NormalizeData(Hep59.1.2.seu)
Hep59.1.2.seu<-ScaleData(Hep59.1.2.seu, features = rownames(Hep59.1.2.seu))
Hep59.1.2.seu <- RenameIdents(object = Hep59.1.2.seu, `1` = "Hypnozoites", `2` = "Schizonts") #hash out if Idents are already renamed
Hep59.1.2.seu.de.markers <- FindMarkers(Hep59.1.2.seu, ident.1 = "Schizonts", ident.2 = "Hypnozoites", logfc.threshold = 0.0)
```

```{r DE testing hypnozoite clusters}
DefaultAssay(Hypnozoites) <- "RNA"
Hypnozoites<-NormalizeData(Hypnozoites)
Hypnozoites<-ScaleData(Hypnozoites, features = rownames(Hypnozoites))
HypnozoiteMarkers<-FindAllMarkers(Hypnozoites, assay = "RNA", test.use = "wilcox", only.pos = T, logfc.threshold = 0.0)
```

Let's save the outputs.
```{r save outputs}
write.table(Hep59.1.2.seu.de.markers, file = "outputs/Schizont_vs_Hypnozoite_DE_markers.txt", append = FALSE, sep = " ", dec = ".",
            row.names = F, col.names = TRUE)
write.table(HypnozoiteMarkers, file = "~/downloads/HypnozoiteMArkers.txt", append = FALSE, sep = " ", dec = ".",
            row.names = F, col.names = TRUE)
```