---
title: "Pv_Hep_scRNAseq_additional_file_6"
author: "Anthony Ruberto"
date: "`r Sys.Date()`"
output:
  html_document: default
  word_document: default
  pdf_document: default
---

Here, we will merge the processed human hepatocyte scRNAseq data from each replicate.

Set options.
```{r set options}
options (future.globals.maxSize = 4000 * 1024^5)
```

Load libraries.
```{r libraries}
library(BUSpaRse)
library(DropletUtils)
library(stringr)
library(tibble)
library(dplyr)
library(ggplot2)
library(scico)
library(scales)
library(rtracklayer)
library(Seurat)
library("leiden")
library(scater)
library(ggpubr)
library(viridis)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)
```

Load Seurat Objects and add metadata.
```{r load Seurat objects}

# processed Human Seurat Objects
Hep51a.seu.hu.f1<-readRDS("outputs/Hep51a.seu.hu.f1.rds")
Hep54a.seu.hu.f1<-readRDS("outputs/Hep54a.seu.hu.f1.rds")
Hep91a.seu.hu.f1<-readRDS("outputs/Hep91a.seu.hu.f1.rds")
Hep94a.seu.hu.f1<-readRDS("outputs/Hep94a.seu.hu.f1.rds")
Hep51b.seu.hu.f1<-readRDS("outputs/Hep51b.seu.hu.f1.rds")
Hep91b.seu.hu.f1<-readRDS("outputs/Hep91b.seu.hu.f1.rds")

# Cells containing parasites - replicate 1
Hep51a.seu.viv.f1.large<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep51a.seu.viv.f1.large.rds")
Hep51a.seu.viv.f1.small<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep51a.seu.viv.f1.small.rds")
Hep91a.seu.viv.f1.small<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep91a.seu.viv.f1.small.rds")

# Cells containing parasites - replicate 2
Hep54a.seu.viv.f1.large<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep54a.seu.viv.f1.large.rds")
Hep54a.seu.viv.f1.small<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep54a.seu.viv.f1.small.rds")
Hep94a.seu.viv.f1.small<-readRDS("~/Documents/Projects/Pv_hepato/src/raw/outputs/Hep94a.seu.viv.f1.small.rds")

# Hypnozoites parsed by inferred status
Hypnozoites.transition<-readRDS( "outputs/Hypnozoites.transition7dec2021.rds")
Hypnozoites.reactivating<- readRDS("outputs/Hypnozoites.reactivating7dec2021.rds")
Hypnozoites.dormant<- readRDS("outputs/Hypnozoites.dormant7dec2021.rds")

# Add Cell IDs. Note: Objects not specified below already contain this metadata.
Hep54a.seu.viv.f1.large$cellID<-rownames(Hep54a.seu.viv.f1.large@meta.data)
Hep54a.seu.viv.f1.small$cellID<-rownames(Hep54a.seu.viv.f1.small@meta.data)
Hypnozoites.dormant$cellID<-rownames(Hypnozoites.dormant@meta.data)
Hypnozoites.reactivating$cellID<-rownames(Hypnozoites.reactivating@meta.data)
Hypnozoites.transition$cellID<-rownames(Hypnozoites.transition@meta.data)
```

Integrate the data.
```{r integrate hepatocyte scRNAseq data}
Hep.all.list <- list(Hep51a.seu.hu.f1, Hep54a.seu.hu.f1, Hep91a.seu.hu.f1,
                     Hep94a.seu.hu.f1, Hep51b.seu.hu.f1, Hep91b.seu.hu.f1) 

# remove the individual Seurat objects after the list is made to clear memory
rm(Hep51a.seu.hu.f1, Hep54a.seu.hu.f1, Hep91a.seu.hu.f1,
   Hep94a.seu.hu.f1, Hep51b.seu.hu.f1, Hep91b.seu.hu.f1)

for (i in 1:length(Hep.all.list)) {
  DefaultAssay(Hep.all.list[[i]]) <- "SCT"
}

features <- SelectIntegrationFeatures(object.list = Hep.all.list, nfeatures = 3000)
Hep.all.list <- PrepSCTIntegration(object.list = Hep.all.list, anchor.features = features)

Hep.anchors <- FindIntegrationAnchors(object.list = Hep.all.list, normalization.method = "SCT",
                                         anchor.features = features, reference = c(5))

hep.combined.sct <- IntegrateData(anchorset = Hep.anchors, normalization.method = "SCT")
```

```{r add feature sets}
hep.combined.sct[["LSECs"]]<- PercentageFeatureSet(hep.combined.sct, features =c("CALCRL", "FCGR2B", "VWF"), assay = 'RNA')
hep.combined.sct[["Hepatocytes"]]<- PercentageFeatureSet(hep.combined.sct, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')
hep.combined.sct[["Cholangiocytes"]]<- PercentageFeatureSet(hep.combined.sct, features =c("KRT19", "EPCAM", "CLDN4", "CLDN10", "SOX9", "MMP7", "CXCL1", "CFTR", "TFF2", "KRT7", "CD24"), assay = 'RNA')
hep.combined.sct[["Hepatic_Stellate_Cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("ACTA2", "COL1A1", "COL1A2", "COL3A1", "DCN", "MYL9"), assay = 'RNA')
hep.combined.sct[["Macrophages"]]<- PercentageFeatureSet(hep.combined.sct, features =c("CD68", "MARCO", "FCGR3A", "LYZ", "PTPRC"), assay = 'RNA')
hep.combined.sct[["ab_T_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("CD2", "CD3D", "TRAC", "IL32", "CD3E", "PTPRC"), assay = 'RNA')
hep.combined.sct[["gd_T_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("NKG7", "FCGR3A", "HOPX", "GNLY", "KLRF1", "CMC1", "CCL3", "PTPRC"), assay = 'RNA')
hep.combined.sct[["NK_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("GZMK", "KLRF1", "CCL3", "CMC1", "NKG7", "PTPRC"), assay = 'RNA')
hep.combined.sct[["Plasma_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("CD27", "IGHG1", "CD79A", "IGHG2", "PTPRC", "IGKC"), assay = 'RNA')
hep.combined.sct[["Mature_B_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("MS4A1", "LTB", "CD52", "IGHD", "CD79A", "PTPRC", "IGKC"), assay = 'RNA')
hep.combined.sct[["Erythroid_cells"]]<- PercentageFeatureSet(hep.combined.sct, features =c("HBB", "SLC25A37", "CA1", "ALAS2"), assay = 'RNA')
hep.combined.sct[["Layers_2_3"]]<- PercentageFeatureSet(hep.combined.sct, features =c("HSD11B1",	"APOM",	"PON3",	"TTC36",	"BCHE",	"F10",	"APOC4-APOC2",	"GOLT1A","ADH1A",	"AQP9",	"G6PC",	"PROX1",	"RCAN1",	"HAMP",	"GPD1",	"CTH",	"TKFC",	"G0S2",	"GHR",	"ABCC2"), assay = 'RNA')
hep.combined.sct[["CV"]]<- PercentageFeatureSet(hep.combined.sct, features =c("BCHE",	"ALDH6A1",	"G6PC","RCAN1"), assay = 'RNA')
hep.combined.sct[["Periportal"]]<- PercentageFeatureSet(hep.combined.sct, features =c("SCD",	"HMGCS1",	"TM7SF2",	"TMEM97",	"CP",	"CRP",	"SLPI",	"ACAT2",	"MSMO1",	"EBP",	"FDFT1",	"SAA2",	"SC5D",	"A2M",	"IFITM2",	"ATF5",	"ITIH3",	"CFHR1",	"C3",	"SPINK1",	"SERPINA1",	"IFITM3",	"FGL1","FGA",	"INSIG1",	"AGT",	"FDPS",	"FABP1",	"IDI1",	"MT1H",	"GC",	"FGB",	"FGG",	"MT2A",	"APOA2"), assay = 'RNA')
hep.combined.sct[["Layer_4"]]<- PercentageFeatureSet(hep.combined.sct, features =c("HPD",	"GSTA1",	"DBI",	"PTGR1",	"HINT1",	"ANGPTL3",	"ADH1C",	"CYB5A","SOD1",	"GPX4",	"GAPDH"), assay = 'RNA')
hep.combined.sct[["PosGNG"]]<- PercentageFeatureSet(hep.combined.sct, features = c(	"ALDOB","ENO1",	"ENO2",	"ENO3",	"FBP1",		"G6PC", "GAPDH",		"MDH1",	"TPI1"), assay, 'RNA')
```

```{r dimensional reduction}
hep.combined.sct <- RunPCA(hep.combined.sct, verbose = FALSE)
hep.combined.sct <- RunUMAP(hep.combined.sct, reduction = "pca", dims = 1:30)
hep.combined.sct <- RunTSNE(hep.combined.sct, reduction = "pca", dims = 1:30)
```

```{r clustering human}
DefaultAssay(hep.combined.sct) <- "integrated"
hep.combined.sct <- FindNeighbors(hep.combined.sct , dims = 1:30, verbose = FALSE)
hep.combined.sct <- FindClusters(hep.combined.sct, verbose = FALSE, resolution = 0.4, algorithm = 1)
```

Save integrated data.
```{r save integrated RDS}
saveRDS(hep.combined.sct, "outputs/hep.combined.sct.rds")
```

```{r differential gene expression analyses human - subset cells}

# Subset cells based on infection status

## Replicate 1, day 5 post-infection, schizonts
subset.hep.infected.day5.large.1 <- subset(hep.combined.sct, cells = Hep51a.seu.viv.f1.large$cellID)
subset.hep.infected.day5.large.1$parasiteID<-"Schizont"

## Replicate 2, day 5 post-infection, schizonts
subset.hep.infected.day5.large.2 <- subset(hep.combined.sct, cells = Hep54a.seu.viv.f1.large$cellID)
subset.hep.infected.day5.large.2$parasiteID<-"Schizont"

## Replicate 1, day 5 post-infection, hypnozoites
subset.hep.infected.day5.small.1 <- subset(hep.combined.sct, cells = Hep51a.seu.viv.f1.small$cellID)
subset.hep.infected.day5.small.1$parasiteID<-"Hypnozoite"

## Replicate 2, day 5 post-infection, hypnozoites
subset.hep.infected.day5.small.2 <- subset(hep.combined.sct, cells = Hep54a.seu.viv.f1.small$cellID)
subset.hep.infected.day5.small.2$parasiteID<-"Hypnozoite"

## Replicate 1, day 9 post-infection, hypnozoites
subset.hep.infected.day9.small.1 <- subset(hep.combined.sct, cells = Hep91a.seu.viv.f1.small$cellID)
subset.hep.infected.day9.small.1$parasiteID<-"Hypnozoite"

## Replicate 12, day 9 post-infection, hypnozoites
subset.hep.infected.day9.small.2 <- subset(hep.combined.sct, cells = Hep94a.seu.viv.f1.small$cellID)
subset.hep.infected.day9.small.2$parasiteID<-"Hypnozoite"

```

```{r differential gene expression analyses human - extract barcode IDs}
# extract cell names (i.e. barcodes) for each of the conditions that will be compared

hep.infected.cells.day5.schizonts<-c(rownames(subset.hep.infected.day5.large.1@meta.data),
                                     rownames(subset.hep.infected.day5.large.2@meta.data))

hep.infected.cells.day5.hypnozoites<-c(rownames(subset.hep.infected.day5.small.1@meta.data),
                                     rownames(subset.hep.infected.day5.small.2@meta.data))

hep.infected.cells.day5.rep1.schizonts<-c(rownames(subset.hep.infected.day5.large.1@meta.data))
hep.infected.cells.day5.rep2.schizonts<-c(rownames(subset.hep.infected.day5.large.2@meta.data))

hep.infected.cells.day5.rep1.hypnozoites<-c(rownames(subset.hep.infected.day5.small.1@meta.data))
hep.infected.cells.day5.rep2.hypnozoites<-c(rownames(subset.hep.infected.day5.small.2@meta.data))


hep.infected.cells.day9.hypnozoites<-c(rownames(subset.hep.infected.day9.small.1@meta.data),
                                       rownames(subset.hep.infected.day9.small.2@meta.data))

hep.infected.cells.day9.rep1.hypnozoites<-c(rownames(subset.hep.infected.day9.small.1@meta.data))
hep.infected.cells.day9.rep2.hypnozoites<-c(rownames(subset.hep.infected.day9.small.2@meta.data))
```

```{r differential gene expression analyses human - DGE analyses 1}

# DEG testing: 

# Identify non-infected cells (Note: this will form Ident.2 for the comparisions of schizont-infected and hypnzoite-infected hepatocytes versus non-infected)

hep.non.infected.naive.exposed.day59<-WhichCells(hep.combined.sct, cells =c(hep.infected.cells.day5.schizonts, hep.infected.cells.day5.hypnozoites, hep.infected.cells.day9.hypnozoites), invert = T)

hep.non.infected.naive.exposed.day59.df <- subset(hep.combined.sct, cells = hep.non.infected.naive.exposed.day59)
```

```{r differential gene expression analyses human - DGE analyses 2}
# related to Table S4A: Schizonts versus non-infected (naive and exposed, days 5 and 9):

hep.combined.sct.de.markers.schizont.v.noninfected <- FindMarkers(hep.combined.sct, 
                                                            ident.1 = hep.infected.cells.day5.schizonts,
                                                            ident.2 = hep.non.infected.naive.exposed.day59,
                                                            logfc.threshold = 0.0, 
                                                            min.pct = 0.25, only.pos = F,
                                                            max.cells.per.ident = 205,
                                                            assay = "RNA")

hep.combined.sct.de.markers.schizont.v.noninfected<-subset(hep.combined.sct.de.markers.schizont.v.noninfected, p_val_adj < 0.05)
hep.combined.sct.de.markers.schizont.v.noninfected<-hep.combined.sct.de.markers.schizont.v.noninfected%>%arrange(desc(avg_log2FC))
```

```{r differential gene expression analyses human - DGE analyses 3}
# related to Table S4B: Schizonts versus non-infected (naive and exposed, days 5 and 9): 

hep.combined.sct.de.markers.hypnozoites.v.noninfected <- FindMarkers(hep.combined.sct, 
                                                                  ident.1 = c(hep.infected.cells.day5.hypnozoites,
                                                                              hep.infected.cells.day9.hypnozoites),
                                                                  ident.2 = hep.non.infected.naive.exposed.day59,
                                                                  logfc.threshold = 0.0, 
                                                                  min.pct = 0.25, only.pos = F,
                                                                  max.cells.per.ident = 924,
                                                                  assay = "RNA")

hep.combined.sct.de.markers.hypnozoites.v.noninfected <-subset(hep.combined.sct.de.markers.hypnozoites.v.noninfected , p_val_adj < 0.05)
hep.combined.sct.de.markers.hypnozoites.v.noninfected <-hep.combined.sct.de.markers.hypnozoites.v.noninfected %>%arrange(desc(avg_log2FC))
```

```{r differential gene expression analyses human - DGE analyses 4}
# related to Table S4E: Schizonts versus non-infected (naive and exposed, days 5 and 9):

hep.non.infected.naive.exposed.day59.df <- SetIdent(hep.non.infected.naive.exposed.day59.df, value = "condition")

hep.non.infected.naive.exposed.day59.de.genes <- FindMarkers(hep.non.infected.naive.exposed.day59.df, 
                                                                  ident.1 = "Infected",
                                                                  ident.2 = "Naive",
                                                                  logfc.threshold = 0.25, 
                                                                  min.pct = 0.25, only.pos = F,
                                                                  max.cells.per.ident = 20731,
                                                                  assay = "RNA")

hep.non.infected.naive.exposed.day59.de.genes <-subset(hep.non.infected.naive.exposed.day59.de.genes , p_val_adj < 0.05)
hep.non.infected.naive.exposed.day59.de.genes <-hep.non.infected.naive.exposed.day59.de.genes %>%arrange(desc(avg_log2FC))
```
