---
title: "Pv_Hep_scRNAseq additional file 1"
author: "Anthony Ruberto"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
---

Post mapping, the data is in its raw format - more specifically, the reads that are due to ambient RNA or associated with a cell are unknown. In this document, we will distinguish between droplets containing cells and ambient RNA.

```{r options}
options (future.globals.maxSize = 4000 * 1024^5)
```

Load libraries.
```{r libraries}
library(BUSpaRse)
library(DropletUtils)
library(stringr)
library(tibble)
library(dplyr)
library(ggplot2)
library(scico)
library(scales)
library(rtracklayer)
library(Seurat)
library("leiden")
library(scater)
library(ggpubr)
library(viridis)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)
library(AnnotationDbi)
library(org.Hs.eg.db)
library(biomaRt)
library(glmGamPoi)
```

Load the gene model files.
```{r load Pv GFF}
pv.gff<- import.gff3("https://plasmodb.org/common/downloads/release-51/PvivaxP01/gff/data/PlasmoDB-51_PvivaxP01.gff") # update with new PlasmoDB updates.

gene.info<-pv.gff %>% 
  mutate(seurat= gsub("_", "-", ID)) %>%
        filter(type =="protein_coding_gene") 

gene.info<-as.data.frame(gene.info)

gene.info$GeneDescription<-paste(gene.info$ID,gene.info$description, sep = "::")

dim(gene.info)
```

```{r load Hs GFF}
tr2g_hs <- tr2g_ensembl("Homo sapiens", ensembl_version = 103, 
                        other_attrs = "description", 
                        gene_biotype_use = "all")

tr2g_hs_filt<-tr2g_hs[,c(1,2,4)]

colnames(tr2g_hs_filt)<-c("transcript", "gene", "description")

tr2g_hs_filt <- apply(tr2g_hs_filt,2,as.character)

tr2g_hs_filt<-as.data.frame(tr2g_hs_filt)

tr2g_hs_filt$gene <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", tr2g_hs_filt$gene) 

rm(tr2g_hs)
```

Load alignment matrices. 
```{r load matrices}
# Day 5 infected, rep 1:
res_mat_1 <- read_count_output("~/path/to/folder/hep51a", name = "genes", tcc = FALSE)
# Day 9 infected, rep 1:
res_mat_2 <- read_count_output("~/path/to/folder/hep91a", name = "genes", tcc = FALSE)
# Day 5 naive hepatocytes control:
res_mat_3 <- read_count_output("~/path/to/folder/hep51b_counts_unfiltered", name = "cells_x_genes", tcc = FALSE)
# Day 9 naive hepatocytes control:
res_mat_4 <- read_count_output("~/path/to/folder/hep91b_counts_unfiltered", name = "cells_x_genes", tcc = FALSE)
# Day 5 infected, rep 2:
res_mat_5 <- read_count_output("~/path/to/folder/hep54a", name = "genes", tcc = FALSE)
# Day 9 infected, rep 2:
res_mat_6 <- read_count_output("~/path/to/folder/Hep94a", name = "genes", tcc = FALSE)
```

Distinguish transcripts in cells from ambient RNA.
```{r filtering reads rep 1 day 5 post infection}
# Replicate 1 - day 5
system.time(
  e.out.res_mat <- emptyDrops(res_mat_1, retain = NULL, lower = 1000)
  )
is.cell.res_mat <- e.out.res_mat$FDR <= 0.001

sum(is.cell.res_mat, na.rm=TRUE)

table(Limited=e.out.res_mat$Limited, Significant=is.cell.res_mat)

plot(e.out.res_mat$Total, -e.out.res_mat$LogProb, col=ifelse(is.cell.res_mat, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_1 <- res_mat_1[, which(e.out.res_mat$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_1)
gene_counts<-Matrix::rowSums(res_mat_1)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_1, "outputs/res_mat_1.rds")
```

```{r filtering reads rep 1 day 9 post infection}
# Replicate 1 - day 9
system.time(
  e.out.res_mat_2 <- emptyDrops(res_mat_2, retain = NULL, lower = 1000)
)

is.cell.res_mat_2 <- e.out.res_mat_2$FDR <= 0.001

sum(is.cell.res_mat_2, na.rm=TRUE)

table(Limited=e.out.res_mat_2$Limited, Significant=is.cell.res_mat_2)

plot(e.out.res_mat_2$Total, -e.out.res_mat_2$LogProb, col=ifelse(is.cell.res_mat_2, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_2 <- res_mat_2[, which(e.out.res_mat_2$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_2)
gene_counts<-Matrix::rowSums(res_mat_2)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_2, "outputs/res_mat_2.rds")
```

```{r filtering reads control day 5 post infection}
# Day 5 - control
system.time(
  e.out.res_mat_3 <- emptyDrops(res_mat_3, retain = NULL, lower = 1000)
)

is.cell.res_mat_3 <- e.out.res_mat_3$FDR <= 0.001

sum(is.cell.res_mat_3, na.rm=TRUE)

table(Limited=e.out.res_mat_3$Limited, Significant=is.cell.res_mat_3)

plot(e.out.res_mat_3$Total, -e.out.res_mat_3$LogProb, col=ifelse(is.cell.res_mat_3, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_3 <- res_mat_3[, which(e.out.res_mat_3$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_3)
gene_counts<-Matrix::rowSums(res_mat_3)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_3, "outputs/res_mat_3.rds")
```

```{r filtering reads control day 9 post infection}
# Day 9 - control
system.time(
  e.out.res_mat_4 <- emptyDrops(res_mat_4, retain = NULL, lower = 1000)
)

is.cell.res_mat_4 <- e.out.res_mat_4$FDR <= 0.001

sum(is.cell.res_mat_4, na.rm=TRUE)

table(Limited=e.out.res_mat_4$Limited, Significant=is.cell.res_mat_4)

plot(e.out.res_mat_4$Total, -e.out.res_mat_4$LogProb, col=ifelse(is.cell.res_mat_4, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_4 <- res_mat_4[, which(e.out.res_mat_4$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_4)
gene_counts<-Matrix::rowSums(res_mat_4)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_4, "outputs/res_mat_4.rds")
```

```{r filtering reads rep 2 day 5 post infection}
# Replicate 2 - day 5
system.time(
  e.out.res_mat_5 <- emptyDrops(res_mat_5, retain = NULL, lower = 1000)
)

is.cell.res_mat_5 <- e.out.res_mat_5$FDR <= 0.001

sum(is.cell.res_mat_5, na.rm=TRUE)

table(Limited=e.out.res_mat_5$Limited, Significant=is.cell.res_mat_5)

plot(e.out.res_mat_5$Total, -e.out.res_mat_5$LogProb, col=ifelse(is.cell.res_mat_5, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_5 <- res_mat_5[, which(e.out.res_mat_5$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_5)
gene_counts<-Matrix::rowSums(res_mat_5)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_5, "outputs/res_mat_5.rds")
```

```{r filtering reads rep 2 day 9 post infection}
# Replicate 2 - day 9
system.time(
  e.out.res_mat_6 <- emptyDrops(res_mat_6, retain = NULL, lower = 1000)
)

is.cell.res_mat_6 <- e.out.res_mat_6$FDR <= 0.001

sum(is.cell.res_mat_6, na.rm=TRUE)

table(Limited=e.out.res_mat_6$Limited, Significant=is.cell.res_mat_6)

plot(e.out.res_mat_6$Total, -e.out.res_mat_6$LogProb, col=ifelse(is.cell.res_mat_6, "red", "black"),
     xlab="Total UMI count", ylab="-Log Probability", log ="y")

res_mat_6 <- res_mat_6[, which(e.out.res_mat_6$FDR<=0.001)]

tot_counts <- Matrix::colSums(res_mat_6)
gene_counts<-Matrix::rowSums(res_mat_6)

summary(tot_counts)
summary(gene_counts)

# Save 
saveRDS(res_mat_6, "outputs/res_mat_6.rds")
```
