---
title: "Pv_Hep_scRNAseq_additional_file_5"
author: "Anthony Ruberto"
date: "`r Sys.Date()`"
output: html_document
---

Here, we will process the data to filter out cells containing reads from P.vivax.

Set options.
```{r set options}
options (future.globals.maxSize = 4000 * 1024^5)
```

Load libraries.
```{r libraries}
library(BUSpaRse)
library(DropletUtils)
library(stringr)
library(tibble)
library(dplyr)
library(ggplot2)
library(scico)
library(scales)
library(rtracklayer)
library(Seurat)
library("leiden")
library(scater)
library(ggpubr)
library(viridis)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)
```

Load the gene model files.
```{r load Hs gene models}
tr2g_hs <- tr2g_ensembl("Homo sapiens", ensembl_version = 103, 
                        other_attrs = "description", 
                        gene_biotype_use = "all")

tr2g_hs_filt<-tr2g_hs[,c(1,2,4)]
colnames(tr2g_hs_filt)<-c("transcript", "gene", "description")
tr2g_hs_filt <- apply(tr2g_hs_filt,2,as.character)
tr2g_hs_filt<-as.data.frame(tr2g_hs_filt)
tr2g_hs_filt$gene <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", tr2g_hs_filt$gene) 
rm(tr2g_hs)
dim(tr2g_hs_filt)
```

Read in RDS files.
```{r read in RDS}
res_mat_1<- readRDS("/path/to/res_mat_1.rds") # Rep 1, day 5 post-infection
res_mat_2<- readRDS("/path/to/res_mat_2.rds") # Rep 1, day 9 post-infection
res_mat_3<- readRDS("/path/to/res_mat_3.rds") # Control, day 5 post-infection
res_mat_4<- readRDS("/path/to/res_mat_4.rds") # Control, day 9 post-infection
res_mat_5<- readRDS("/path/to/res_mat_5.rds") # Rep 2, day 5 post-infection
res_mat_6<- readRDS("/path/to/res_mat_6.rds") # Rep 2, day 9 post-infection

# P.vivax processed Seurat Objects
readRDS("outputs/Hep51a.seu.viv.f1.large.rds")
readRDS("outputs/Hep51a.seu.viv.f1.small.rds")
readRDS("outputs/Hep91a.seu.viv.f1.small.rds")

readRDS("outputs/Hep54a.seu.viv.f1.large.rds")
readRDS("outputs/Hep54a.seu.viv.f1.small.rds")
readRDS("outputs/Hep94a.seu.viv.f1.small.rds")
```

The row names of the matrices are currently annotated using ENSEMBL codes (i.e. ENSGXXXXXXXXXXX.X).
Let's switch these to Gene Symbols.
```{r ENSEMBL to gene symbol}
# replicate 1, day 5 post infection:
rownames(res_mat_1)
## Remove version number
rownames(res_mat_1) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_1))
res_mat_human<-res_mat_1[rownames(res_mat_1) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_human) <- tr2g_hs_filt$description[match(rownames(res_mat_human), tr2g_hs_filt$gene)]

# replicate 1, day 9 post infection:
rownames(res_mat_2)
## Remove version number
rownames(res_mat_2) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_2)) 
res_mat_2<-res_mat_2[rownames(res_mat_2) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_2) <- tr2g_hs_filt$description[match(rownames(res_mat_2), tr2g_hs_filt$gene)]

# control 1 (naive), day 5 post infection:
rownames(res_mat_3)
## Remove version number
rownames(res_mat_3) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_3)) 
res_mat_3<-res_mat_3[rownames(res_mat_3) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_3) <- tr2g_hs_filt$description[match(rownames(res_mat_3), tr2g_hs_filt$gene)]

# control 2 (naive), day 9 post-infection:
rownames(res_mat_4)
## Remove version number
rownames(res_mat_4) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_4)) 
res_mat_4<-res_mat_4[rownames(res_mat_4) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_4) <- tr2g_hs_filt$description[match(rownames(res_mat_4), tr2g_hs_filt$gene)]

# replicate 2, day 5 post infection:
rownames(res_mat_5)
## Remove version number
rownames(res_mat_5) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_5)) 
res_mat_5<-res_mat_5[rownames(res_mat_5) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_5) <- tr2g_hs_filt$description[match(rownames(res_mat_5), tr2g_hs_filt$gene)]

# replicate 2, day 9 post infection:
rownames(res_mat_6)
## Remove version number
rownames(res_mat_6) <- gsub("(ENSG[0-9]+)\\.[0-9]+", "\\1", rownames(res_mat_6)) 
res_mat_6<-res_mat_6[rownames(res_mat_6) %in% tr2g_hs_filt$gene, ]
rownames(res_mat_6) <- tr2g_hs_filt$description[match(rownames(res_mat_6), tr2g_hs_filt$gene)]

```

Create Seurat Objects.
```{r create Seurat objects Human}
# replicate 1, day 5 post infection:
Hep51a.seu.hu<- CreateSeuratObject(res_mat_human, min.cells = 30)
Hep51a.seu.hu<- AddMetaData(Hep51a.seu.hu, metadata = cell_species.Hep51a$species, 
                         col.name = "species")
Hep51a.seu.hu$day<-"Five"
Hep51a.seu.hu$replicate<-"1"
Hep51a.seu.hu$condition<-"Infected"
Hep51a.seu.hu$uniqueID<-"Day5_Rep1_Infected"
Hep51a.seu.hu<-RenameCells(Hep51a.seu.hu, add.cell.id = "51inf")
Hep51a.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep51a.seu.hu, pattern = "^MT-")
Hep51a.seu.hu$is_outlier<-isOutlier(metric = Hep51a.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE) # plot to visualize outlier cells
Hep51a.seu.hu.f1<-Hep51a.seu.hu[,Hep51a.seu.hu$nFeature_RNA >750 ] 
Hep51a.seu.hu.f1$is_outlier<-NULL
Hep51a.seu.hu.f1$is_outlier<-isOutlier(metric = Hep51a.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE) # plot to viz post filtering
Hep51a.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep51a.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep51a.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep51a.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')

# replicate 1, day 9 post infection:
Hep91a.seu.hu<- CreateSeuratObject(res_mat_2, min.cells = 30)
Hep91a.seu.hu<- AddMetaData(Hep91a.seu.hu, metadata = cell_species.Hep91a$species, 
                            col.name = "species")
Hep91a.seu.hu$day<-"Nine"
Hep91a.seu.hu$replicate<-"1"
Hep91a.seu.hu$condition<-"Infected"
Hep91a.seu.hu$uniqueID<-"Day9_Rep1_Infected"
Hep91a.seu.hu<-RenameCells(Hep91a.seu.hu, add.cell.id = "91inf")
Hep91a.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep91a.seu.hu, pattern = "^MT-")
Hep91a.seu.hu$is_outlier<-isOutlier(metric = Hep91a.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep91a.seu.hu.f1<-Hep91a.seu.hu[,Hep91a.seu.hu$nFeature_RNA >750]
Hep91a.seu.hu.f1$is_outlier<-NULL
Hep91a.seu.hu.f1$is_outlier<-isOutlier(metric = Hep91a.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep91a.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep91a.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep91a.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep91a.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')

# control 1 (naive), day 5 post infection:
Hep51b.seu.hu<- CreateSeuratObject(res_mat_3, min.cells = 30)
Hep51b.seu.hu<- AddMetaData(Hep51b.seu.hu, metadata = cell_species.Hep51b$species, 
                            col.name = "species")
Hep51b.seu.hu$day<-"Five"
Hep51b.seu.hu$replicate<-"1"
Hep51b.seu.hu$condition<-"Naive"
Hep51b.seu.hu$uniqueID<-"Day5_Rep1_Naive"
Hep51b.seu.hu<-RenameCells(Hep51b.seu.hu, add.cell.id = "51naive")
Hep51b.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep51b.seu.hu, pattern = "^MT-")
Hep51b.seu.hu$is_outlier<-isOutlier(metric = Hep51b.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep51b.seu.hu.f1<-Hep51b.seu.hu[,Hep51b.seu.hu$nFeature_RNA >750]
Hep51b.seu.hu.f1$is_outlier<-NULL
Hep51b.seu.hu.f1$is_outlier<-isOutlier(metric = Hep51b.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep51b.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep51b.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep51b.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep51b.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')

# control 2 (naive), day 9 post infection:
Hep91b.seu.hu<- CreateSeuratObject(res_mat_4, min.cells = 30)
Hep91b.seu.hu<- AddMetaData(Hep91b.seu.hu, metadata = cell_species.Hep91b$species, 
                            col.name = "species")
Hep91b.seu.hu$day<-"Nine"
Hep91b.seu.hu$replicate<-"1"
Hep91b.seu.hu$condition<-"Naive"
Hep91b.seu.hu$uniqueID<-"Day9_Rep1_Naive"
Hep91b.seu.hu<-RenameCells(Hep91b.seu.hu, add.cell.id = "91naive")
Hep91b.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep91b.seu.hu, pattern = "^MT-")
Hep91b.seu.hu$is_outlier<-isOutlier(metric = Hep91b.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep91b.seu.hu.f1<-Hep91b.seu.hu[,Hep91b.seu.hu$nFeature_RNA >750] 
Hep91b.seu.hu.f1$is_outlier<-NULL
Hep91b.seu.hu.f1$is_outlier<-isOutlier(metric = Hep91b.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep91b.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep91b.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep91b.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep91b.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')

# replicate 2, day 5 post infection:
Hep54a.seu.hu<- CreateSeuratObject(res_mat_5, min.cells = 30)
Hep54a.seu.hu<- AddMetaData(Hep54a.seu.hu, metadata = cell_species.Hep54a$species, 
                            col.name = "species")
Hep54a.seu.hu$day<-"Five"
Hep54a.seu.hu$replicate<-"2"
Hep54a.seu.hu$condition<-"Infected"
Hep54a.seu.hu$uniqueID<-"Day5_Rep2_Infected"
Hep54a.seu.hu<-RenameCells(Hep54a.seu.hu, add.cell.id = "52inf")
Hep54a.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep54a.seu.hu, pattern = "^MT-")
Hep54a.seu.hu$is_outlier<-isOutlier(metric = Hep54a.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep54a.seu.hu.f1<-Hep54a.seu.hu[,Hep54a.seu.hu$nFeature_RNA >750]
Hep54a.seu.hu.f1$is_outlier<-NULL
Hep54a.seu.hu.f1$is_outlier<-isOutlier(metric = Hep54a.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep54a.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep54a.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep54a.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep54a.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "PCK1", "AFP", "BCHE"), assay = 'RNA')

# replicate 2, day 9 post infection:
Hep94a.seu.hu<- CreateSeuratObject(res_mat_6, min.cells = 30)
Hep94a.seu.hu<- AddMetaData(Hep94a.seu.hu, metadata = cell_species.Hep94a$species, 
                            col.name = "species")
Hep94a.seu.hu$day<-"Nine"
Hep94a.seu.hu$replicate<-"2"
Hep94a.seu.hu$condition<-"Infected"
Hep94a.seu.hu$uniqueID<-"Day9_Rep2_Infected"
Hep94a.seu.hu<-RenameCells(Hep94a.seu.hu, add.cell.id = "92inf")
Hep94a.seu.hu[["percent.mt"]] <- PercentageFeatureSet(Hep94a.seu.hu, pattern = "^MT-")
Hep94a.seu.hu$is_outlier<-isOutlier(metric = Hep94a.seu.hu$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep94a.seu.hu.f1<-Hep94a.seu.hu[,Hep94a.seu.hu$nFeature_RNA >750] 
Hep94a.seu.hu.f1$is_outlier<-NULL
Hep94a.seu.hu.f1$is_outlier<-isOutlier(metric = Hep94a.seu.hu.f1$nCount_RNA,nmads = 3, type = "both", log = TRUE)
Hep94a.seu.hu.f1[["Alb"]]<- PercentageFeatureSet(Hep94a.seu.hu.f1, features=c("ALB"), pattern = NULL, assay = 'RNA')
Hep94a.seu.hu.f1[["Hepatocytes"]]<- PercentageFeatureSet(Hep94a.seu.hu.f1, features =c("ALB", "HAMP", "ARG1", "AFP", "BCHE"), assay = 'RNA')
```

Data normalization, transformation, and clustering.
```{r data norm, transformation, clustering human}
# replicate 1, day 5 post infection:
Hep51a.seu.hu.f1 <- SCTransform(Hep51a.seu.hu.f1, 
                             variable.features.n = NULL, 
                             variable.features.rv.th = 1.3)
Hep51a.seu.hu.f1  <- RunPCA(Hep51a.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep51a.seu.hu.f1)
Hep51a.seu.hu.f1 <- RunUMAP(Hep51a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep51a.seu.hu.f1 <- RunTSNE(Hep51a.seu.hu.f1, verbose = FALSE)
Hep51a.seu.hu.f1 <- FindNeighbors(Hep51a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep51a.seu.hu.f1 <- FindClusters(Hep51a.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)

# replicate 2, day 9 post infection:
Hep91a.seu.hu.f1 <- SCTransform(Hep91a.seu.hu.f1, 
                                variable.features.n = NULL, 
                                variable.features.rv.th = 1.3)
Hep91a.seu.hu.f1  <- RunPCA(Hep91a.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep91a.seu.hu.f1)
Hep91a.seu.hu.f1 <- RunUMAP(Hep91a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep91a.seu.hu.f1 <- RunTSNE(Hep91a.seu.hu.f1, verbose = FALSE)
Hep91a.seu.hu.f1 <- FindNeighbors(Hep91a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep91a.seu.hu.f1 <- FindClusters(Hep91a.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)

# control 1 (naive), day 5 post infection
Hep51b.seu.hu.f1 <- SCTransform(Hep51b.seu.hu.f1, 
                                variable.features.n = NULL, 
                                variable.features.rv.th = 1.3)
Hep51b.seu.hu.f1  <- RunPCA(Hep51b.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep51b.seu.hu.f1)
Hep51b.seu.hu.f1 <- RunUMAP(Hep51b.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep51b.seu.hu.f1 <- RunTSNE(Hep51b.seu.hu.f1, verbose = FALSE)
Hep51b.seu.hu.f1 <- FindNeighbors(Hep51b.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep51b.seu.hu.f1 <- FindClusters(Hep51b.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)

# control 2 (naive), day 9 post infection
Hep91b.seu.hu.f1 <- SCTransform(Hep91b.seu.hu.f1, 
                                variable.features.n = NULL, 
                                variable.features.rv.th = 1.3)
Hep91b.seu.hu.f1  <- RunPCA(Hep91b.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep91b.seu.hu.f1)
Hep91b.seu.hu.f1 <- RunUMAP(Hep91b.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep91b.seu.hu.f1 <- RunTSNE(Hep91b.seu.hu.f1, verbose = FALSE)
Hep91b.seu.hu.f1 <- FindNeighbors(Hep91b.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep91b.seu.hu.f1 <- FindClusters(Hep91b.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)

# replicate 2, day 5 post infection:
Hep54a.seu.hu.f1 <- SCTransform(Hep54a.seu.hu.f1, 
                                variable.features.n = NULL, 
                                variable.features.rv.th = 1.3)
Hep54a.seu.hu.f1  <- RunPCA(Hep54a.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep54a.seu.hu.f1)
Hep54a.seu.hu.f1 <- RunUMAP(Hep54a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep54a.seu.hu.f1 <- RunTSNE(Hep54a.seu.hu.f1, verbose = FALSE)
Hep54a.seu.hu.f1 <- FindNeighbors(Hep54a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep54a.seu.hu.f1 <- FindClusters(Hep54a.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)

# replcate 2, day 9 post infection:
Hep94a.seu.hu.f1 <- SCTransform(Hep94a.seu.hu.f1, 
                                variable.features.n = NULL, 
                                variable.features.rv.th = 1.3)

# Dimension reduction
Hep94a.seu.hu.f1  <- RunPCA(Hep94a.seu.hu.f1 , verbose = FALSE)
ElbowPlot(Hep94a.seu.hu.f1)
Hep94a.seu.hu.f1 <- RunUMAP(Hep94a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep94a.seu.hu.f1 <- RunTSNE(Hep94a.seu.hu.f1, verbose = FALSE)
DimPlot(Hep94a.seu.hu.f1, reduction = "tsne")+border()+NoLegend()
Hep94a.seu.hu.f1 <- FindNeighbors(Hep94a.seu.hu.f1 , dims = 1:30, verbose = FALSE)
Hep94a.seu.hu.f1 <- FindClusters(Hep94a.seu.hu.f1, verbose = FALSE, resolution = 0.5, algorithm = 4)
```

We can now visualize the data in low dimensional space. Note that since we processed the parasite data,
we can highlight the cells that contain parasite RNA.
```{r visualize objects in low dim space}
ggarrange(
  DimPlot(Hep51a.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep51a.seu.hu.f1, label = F, cells.highlight = Hep51a.seu.viv.f1.large$cellID, reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep51a.seu.hu.f1, label = F, cells.highlight = Hep51a.seu.viv.f1.small$cellID, reduction = "tsne") + NoLegend()+border(),
  align = "hv", ncol = 3, nrow = 1)

ggarrange(
  DimPlot(Hep91a.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep91a.seu.hu.f1, label = F,  cells.highlight = Hep91a.seu.viv.f1$cellID, reduction = "tsne") + NoLegend()+border(),
  align = "hv", ncol = 2, nrow = 2
)

DimPlot(Hep51b.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border()

DimPlot(Hep91b.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border()

ggarrange(
  DimPlot(Hep54a.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep54a.seu.hu.f1, label = F, cells.highlight = Hep54a.seu.viv.f1.large$cellID, reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep54a.seu.hu.f1, label = F, cells.highlight = Hep54a.seu.viv.f1.small$cellID, reduction = "tsne") + NoLegend()+border(),
  align = "hv", ncol = 3, nrow = 1)

ggarrange(
  DimPlot(Hep94a.seu.hu.f1, label = TRUE, group.by = "seurat_clusters", reduction = "tsne") + NoLegend()+border(),
  DimPlot(Hep94a.seu.hu.f1, label = F, cells.highlight = Hep94a.seu.viv.f1$cellID, reduction = "tsne", order = T) + NoLegend()+border(),
  align = "hv", ncol = 3, nrow = 1)
```

Save RDS files.
```{r save RDS files}
saveRDS(Hep51a.seu.hu.f1, "outputs/Hep51a.seu.hu.f1.rds")
saveRDS(Hep91a.seu.hu.f1, "outputs/Hep91a.seu.hu.f1.rds")
saveRDS(Hep51b.seu.hu.f1, "outputs/Hep51b.seu.hu.f1.rds")
saveRDS(Hep91b.seu.hu.f1, "outputs/Hep91b.seu.hu.f1.rds")
saveRDS(Hep54a.seu.hu.f1, "outputs/Hep54a.seu.hu.f1.rds")
saveRDS(Hep94a.seu.hu.f1, "outputs/Hep94a.seu.hu.f1.rds")
```