---
title: "Pv_Hep_scRNAseq_additional_file_2"
author: "Anthony Ruberto"
date: "`r Sys.Date()`"
output: html_document
---

Here, we will process the data to filter out cells containing reads from host hepatocytes.

Load libraries.
```{r libraries}
library(BUSpaRse)
library(DropletUtils)
library(stringr)
library(tibble)
library(dplyr)
library(ggplot2)
library(scico)
library(scales)
library(rtracklayer)
library(Seurat)
library("leiden")
library(scater)
library(ggpubr)
library(viridis)
library(EnhancedVolcano)
library(pheatmap)
library(RColorBrewer)
```

Load the gene model files.
```{r load Pv GFF}
pv.gff<- import.gff3("https://plasmodb.org/common/downloads/release-51/PvivaxP01/gff/data/PlasmoDB-51_PvivaxP01.gff") # update with new PlasmoDB updates.

gene.info<-pv.gff %>% 
  mutate(seurat= gsub("_", "-", ID)) %>%
        filter(type =="protein_coding_gene") 

gene.info<-as.data.frame(gene.info)
gene.info$GeneDescription<-paste(gene.info$ID,gene.info$description, sep = "::")
dim(gene.info)
```

Read in RDS files.
```{r read in RDS}
res_mat_1<- readRDS("outputs/res_mat_1.rds") # Rep 1, day 5 post-infection
res_mat_2<- readRDS("outputs/res_mat_2.rds") # Rep 1, day 9 post-infection
res_mat_5<- readRDS("outputs/res_mat_5.rds") # Rep 2, day 5 post-infection
res_mat_6<- readRDS("outputs/res_mat_6.rds") # Rep 2, day 9 post-infection
```

Keep only P.vivax transcripts (Filter out human transcripts).
```{r Pv only}
res_mat_1_vivax<-res_mat_1[rownames(res_mat_1) %in% gene.info$ID, ]
res_mat_2_vivax<-res_mat_2[rownames(res_mat_2) %in% gene.info$ID, ]
res_mat_5_vivax<-res_mat_5[rownames(res_mat_5) %in% gene.info$ID, ]
res_mat_6_vivax<-res_mat_6[rownames(res_mat_6) %in% gene.info$ID, ]
```

Create Seurat objects.
```{r Seurat object P.vivax}
Hep51a.seu.viv<- CreateSeuratObject(res_mat_1_vivax)
Hep91a.seu.viv<- CreateSeuratObject(res_mat_2_vivax)
Hep54a.seu.viv<- CreateSeuratObject(res_mat_5_vivax)
Hep94a.seu.viv<- CreateSeuratObject(res_mat_6_vivax)
```

Add metadata.
```{r add metadata}

# Replicate 1 - Day 5
Hep51a.seu.viv$day<-"Five"
Hep51a.seu.viv$replicate<-"1"
Hep51a.seu.viv$condition<-"Infected"
Hep51a.seu.viv$uniqueID<-"Day5_Rep1_Infected"
Hep51a.seu.viv<-RenameCells(Hep51a.seu.viv, add.cell.id = "51inf")

# Replicate 1 - Day 9
Hep91a.seu.viv$day<-"Nine"
Hep91a.seu.viv$replicate<-"1"
Hep91a.seu.viv$condition<-"Infected"
Hep91a.seu.viv$uniqueID<-"Day9_Rep1_Infected"
Hep91a.seu.viv<-RenameCells(Hep91a.seu.viv, add.cell.id = "91inf")

# Replicate 2 - Day 5
Hep54a.seu.viv$day<-"Five"
Hep54a.seu.viv$replicate<-"2"
Hep54a.seu.viv$condition<-"Infected"
Hep54a.seu.viv$uniqueID<-"Day5_Rep2_Infected"
Hep54a.seu.viv<-RenameCells(Hep54a.seu.viv, add.cell.id = "52inf")

# Relicate 2 - Day 9
Hep94a.seu.viv$day<-"Nine"
Hep94a.seu.viv$replicate<-"2"
Hep94a.seu.viv$condition<-"Infected"
Hep94a.seu.viv$uniqueID<-"Day9_Rep2_Infected"
Hep94a.seu.viv<-RenameCells(Hep94a.seu.viv, add.cell.id = "92inf")
```

Filter out cells with low transcript diversity.
```{r remove cells with low transcript diversity}
Hep51a.seu.viv.f1<-Hep51a.seu.viv[,Hep51a.seu.viv$nFeature_RNA >60]
Hep91a.seu.viv.f1<-Hep91a.seu.viv[,Hep91a.seu.viv$nFeature_RNA >60]
Hep54a.seu.viv.f1<-Hep54a.seu.viv[,Hep54a.seu.viv$nFeature_RNA >60]
Hep94a.seu.viv.f1<-Hep94a.seu.viv[,Hep94a.seu.viv$nFeature_RNA >60]
```

Filter out transcripts with low expression.
```{r remove transcipts with low expression}
# Replicate 1 - Day 5
counts <- GetAssayData(object = Hep51a.seu.viv.f1, slot = "counts")
nonzero <- counts > 0
keep_genes <- Matrix::rowSums(nonzero) >= 10
filtered_counts <- counts[keep_genes, ]
# create new Seurat object with cleaned dataset
Hep51a.seu.viv.f1 <- CreateSeuratObject(filtered_counts, meta.data = Hep51a.seu.viv.f1@meta.data)

# Replicate 1 - Day 9
counts <- GetAssayData(object = Hep91a.seu.viv.f1, slot = "counts")
nonzero <- counts > 0
keep_genes <- Matrix::rowSums(nonzero) >= 10
filtered_counts <- counts[keep_genes, ]
# create new Seurat object with cleaned dataset
Hep91a.seu.viv.f1 <- CreateSeuratObject(filtered_counts, meta.data = Hep91a.seu.viv.f1@meta.data)

# Replicate 2 - Day 5
Hep54a.seu.viv.f1
counts <- GetAssayData(object = Hep54a.seu.viv.f1, slot = "counts")
nonzero <- counts > 0
keep_genes <- Matrix::rowSums(nonzero) >= 10
filtered_counts <- counts[keep_genes, ]
# create new Seurat object with cleaned dataset
Hep54a.seu.viv.f1 <- CreateSeuratObject(filtered_counts, meta.data = Hep54a.seu.viv.f1@meta.data)
```

Data normalization, transformation, and clustering.
```{r sctransform and clustering}
# Replicate 1 - Day 5
Hep51a.seu.viv.f1 <- SCTransform(Hep51a.seu.viv.f1, variable.features.n = NULL, 
                                 variable.features.rv.th = 1.3)
Hep51a.seu.viv.f1 <- RunPCA(Hep51a.seu.viv.f1, verbose = FALSE)
Hep51a.seu.viv.f1 <- RunUMAP(Hep51a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep51a.seu.viv.f1 <- RunTSNE(Hep51a.seu.viv.f1, dims = 1:20, perplexity = 20)
Hep51a.seu.viv.f1 <- FindNeighbors(Hep51a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep51a.seu.viv.f1 <- FindClusters(Hep51a.seu.viv.f1, verbose = FALSE, 
                                  resolution = 0.1, algorithm = 4)

# Replicate 1 - Day 9
Hep91a.seu.viv.f1 <- SCTransform(Hep91a.seu.viv.f1, variable.features.n = NULL, 
                                 variable.features.rv.th = 1.3)
Hep94a.seu.viv.f1 <- RunPCA(Hep94a.seu.viv.f1, verbose = FALSE)
Hep91a.seu.viv.f1 <- RunUMAP(Hep91a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep91a.seu.viv.f1 <- RunTSNE(Hep91a.seu.viv.f1, dims = 1:20, perplexity = 20)
Hep91a.seu.viv.f1 <- FindNeighbors(Hep91a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep91a.seu.viv.f1 <- FindClusters(Hep91a.seu.viv.f1, verbose = FALSE, 
                                  resolution = 0.1, algorithm = 4)

# Replicate 2 - Day 5
Hep54a.seu.viv.f1 <- SCTransform(Hep54a.seu.viv.f1, variable.features.n = NULL, 
                                 variable.features.rv.th = 1.3)
Hep54a.seu.viv.f1 <- RunPCA(Hep54a.seu.viv.f1, verbose = FALSE)
Hep54a.seu.viv.f1 <- RunUMAP(Hep54a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep54a.seu.viv.f1 <- RunTSNE(Hep54a.seu.viv.f1, dims = 1:20, perplexity = 20)
Hep54a.seu.viv.f1 <- FindNeighbors(Hep54a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep54a.seu.viv.f1 <- FindClusters(Hep54a.seu.viv.f1, verbose = FALSE, 
                                  resolution = 0.1, algorithm = 4)

# Replicate 2 - Day 9
Hep94a.seu.viv.f1 <- SCTransform(Hep94a.seu.viv.f1, variable.features.n = NULL, 
                                 variable.features.rv.th = 1.3)
Hep94a.seu.viv.f1 <- RunPCA(Hep94a.seu.viv.f1, verbose = FALSE)
Hep94a.seu.viv.f1 <- RunUMAP(Hep94a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep94a.seu.viv.f1 <- RunTSNE(Hep94a.seu.viv.f1, dims = 1:20, perplexity = 20)
Hep94a.seu.viv.f1 <- FindNeighbors(Hep94a.seu.viv.f1 , dims = 1:20, verbose = FALSE)
Hep94a.seu.viv.f1 <- FindClusters(Hep94a.seu.viv.f1, verbose = FALSE, 
                                  resolution = 0.1, algorithm = 4)
```

Subset schizonts and hypnozoites.
```{r subset liver forms}
# Replicate 1 - Day 5
Hep51a.seu.viv.f1 <- RenameIdents(object = Hep51a.seu.viv.f1, 
                                  `1` = "Hypnozoites", `2` = "Schizonts")
Hep51a.seu.viv.f1.large <- subset(Hep51a.seu.viv.f1, idents = "Schizonts")
Hep51a.seu.viv.f1.small <- subset(Hep51a.seu.viv.f1, idents = "Hypnozoites")
Hep51a.seu.viv.f1.large$cellID <- rownames(Hep51a.seu.viv.f1.large@meta.data)
Hep51a.seu.viv.f1.small$cellID <- rownames(Hep51a.seu.viv.f1.small@meta.data)

# Replicate 1 - Day 9
Hep91a.seu.viv.f1.small<-Hep91a.seu.viv.f1
Hep91a.seu.viv.f1.small$cellID<-rownames(Hep91a.seu.viv.f1.small@meta.data)

# Replicate 2 - Day 5
Hep54a.seu.viv.f1 <- RenameIdents(object = Hep54a.seu.viv.f1, 
                                  `1` = "Hypnozoites", `2` = "Schizonts")
Hep54a.seu.viv.f1.large <- subset(Hep54a.seu.viv.f1, idents = "Schizonts")
Hep54a.seu.viv.f1.small <- subset(Hep54a.seu.viv.f1, idents = "Hypnozoites")
Hep54a.seu.viv.f1.large$cellID <- rownames(Hep54a.seu.viv.f1.large@meta.data)
Hep54a.seu.viv.f1.small$cellID <- rownames(Hep54a.seu.viv.f1.small@meta.data)

# Replicate 2 - Day 9
Hep94a.seu.viv.f1.small<-Hep94a.seu.viv.f1
Hep94a.seu.viv.f1.small$cellID<-rownames(Hep94a.seu.viv.f1.small@meta.data)
```

Save RDS files.
```{r save RDS files}
saveRDS(Hep51a.seu.viv.f1, "outputs/Hep51a.seu.viv.f1.rds")
saveRDS(Hep91a.seu.viv.f1, "outputs/Hep91a.seu.viv.f1.rds")
saveRDS(Hep54a.seu.viv.f1, "outputs/Hep54a.seu.viv.f1.rds")
saveRDS(Hep94a.seu.viv.f1, "outputs/Hep94a.seu.viv.f1.rds")

saveRDS(Hep51a.seu.viv.f1.large, "outputs/Hep51a.seu.viv.f1.large.rds")
saveRDS(Hep51a.seu.viv.f1.small, "outputs/Hep51a.seu.viv.f1.small.rds")
saveRDS(Hep91a.seu.viv.f1.small, "outputs/Hep91a.seu.viv.f1.small.rds")

saveRDS(Hep54a.seu.viv.f1.large, "outputs/Hep54a.seu.viv.f1.large.rds")
saveRDS(Hep54a.seu.viv.f1.small, "outputs/Hep54a.seu.viv.f1.small.rds")
saveRDS(Hep94a.seu.viv.f1.small, "outputs/Hep94a.seu.viv.f1.small.rds")
```

